app/models/spree/return_authorization.rb

```diff
@@ -64,6 +64,11 @@ def returnable_inventory
       order.shipped_shipments.collect{ |s| s.inventory_units.to_a }.flatten
     end
 
+    # Used when Adjustment#update! wants to update the related adjustment
+    def compute_amount(*args)
```

Comment:
Override? Or was this not defined before?

Reply:
It's an override. The method is defined initially in `Spree::Core::CalculatedAdjustments` and overridden here because the behavior needs to be slightly different in this case, eg: most of the time adjustments call an `originator` which has a `calculator` object that applies some kind of calculation to determine the amount, but in this specific case this type of adjustment does not have an associated calculator and is always a flat amount.

It's added to this class in spree/2-2-stable [here](https://github.com/spree/spree/blob/241f1c69d629a86dd72b6df17f805eda0bae8444/core/app/models/spree/return_authorization.rb#L71-L75)

Reply:
Ah, I missed a bit that's needed here. Good catch!

Reply:
Now I see the connection! :smile: